imageio/imageio-psd/src/main/java/com/twelvemonkeys/imageio/plugins/psd/PSDLayerInfo.java

```diff
@@ -144,7 +153,15 @@
                     }
                     layerId = pInput.readInt();
                     break;
-
+                case PSD.lsct:
+                    sectionDividerSettingType = SectionDividerSetting.fromInt(pInput.readInt());
+                    if (resourceLength >= 16) {
+                        pInput.skipBytes(12);
+                    }
+                    else if (resourceLength >= 12) {
+                        pInput.skipBytes(8);
+                    }
```

Comment:
I believe the following would be both simpler to understand and more correct (as we read 4 bytes in the `readInt()` method):

```suggestion
                    pInput.skipBytes(resourceLength - 4);
```